### DISCUSSION TITLE
Battery Setup

### DISCUSSION THREAD

**Discussion by fboundy** [alienatedsec/solis-ha-modbus-cloud#12] [Q&A]
Can you please share a little more information on how you have set up the battery with the 3rd Waveshare device? Is that also linked into Home Assistant?

**Comment by alienatedsec** [comment]
@fboundy it's still on my to-do list. There will be a need for creating a new HA integration or playing with Waveshare's MQTT. I got the cable disconnected for now as I had occasional CAN failure alarms on the inverter. Still not sure what caused it, but feels like the battery or the inverter was unhappy without that termination RJ45 or with constant RS485 connections on PINs 2 and 3. That is despite the RJ45 for my waveshare also including a 120 ohms termination resistor on PINs 4 and 5. 

Regardless, that was the early stage of the project with an old inverter firmware. Having 470044 installed and all being stable for a while, I could start looking into battery monitoring all over again. Who knows, it could be a joint effort(?)

What is your idea?

https://github.com/alienatedsec/solis-ha-modbus-cloud#future-work

**Reply by alienatedsec** [reply]
@fboundy any thoughts?
